tendrl/node_agent/objects/message/__init__.py

```diff
@@ -0,0 +1,24 @@
+from tendrl.commons import etcdobj
+from tendrl.commons.message import Message as message
+from tendrl.node_agent import objects
+
+
+class Message(message, objects.NodeAgentBaseObject):
+    def __init__(self, **message_arg):
+        super(Message, self).__init__(**message_arg)
+        self.value = 'Messages/%s'
+        self._etcd_cls = _MessageEtcd
+
+
+class _MessageEtcd(etcdobj.EtcdObj):
+    """Message object, lazily updated
+
+    """
+    __name__ = 'Messages/%s'
+    _tendrl_cls = Message
+
+    def render(self):
+        self.__name__ = self.__name__ % (
+            self.message_id
```

Comment:
@r0h4n 

class Message(message, objects.NodeAgentBaseObject):
    def __init__(self, **message_arg):
        super(Message, self).__init__(**message_arg)

class Message(message, objects.NodeAgentBaseObject) it works like Message inherited from message and  message inherited from  objects.NodeAgentBaseObject

so Message inherited from Message(common) so **message is send to commons Message class 
and common message you can see in __init__    super(Message, self).__init__(*args, **kwargs)
so it will call objects.NodeAgentBaseObject constructor... so  self.message_id accessible by Message class

it work i tested